### PR TITLE
[Rust] Update documentation

### PR DESCRIPTION
**Pull request by Ryan-J-D**
Adjusted existing hooks and added all of the new ones added since Oxide build 2.0.3187 (Apr 13th)

### CODE REVIEW

File: source/includes/rust/hooks_item.md [hunk 1]
```diff
@@ -48,6 +48,30 @@ void CanStackItem(Item item, Item targetItem)
...
+
+``` csharp
+void OnMaxStackable(Item item, Item targetItem)
+{
+    Puts("OnMaxStackable works!");
+}
+```
+
+ * Called when an items max stackable is calculated
+ * Returning a non-null value overrides default behavior
```
Comment (Mughisi) [inline source/includes/rust/hooks_item.md]: Returning a numeric value (int) overrides the default value.

File: source/includes/rust/hooks_player.md [hunk 2]
```diff
@@ -351,6 +351,7 @@ void OnPlayerRespawn(BasePlayer player)
 
  * Called when the player is attempting to respawn
  * Returning a non-null value overrides the default behavior
+ * Returning a Vector3 position changes the players spawnpoint
```
Comment (Mughisi) [inline source/includes/rust/hooks_player.md]: Returning a BasePlayer.SpawnPoint (takes a position and rotation) overrides the respawn location.

File: source/includes/rust/hooks_player.md [hunk 3]
```diff
@@ -351,6 +351,7 @@ void OnPlayerRespawn(BasePlayer player)
 
  * Called when the player is attempting to respawn
  * Returning a non-null value overrides the default behavior
```
Comment (Mughisi) [inline source/includes/rust/hooks_player.md]: Only returning a BasePlayer.SpawnPoint will override the default behavior, anything else will just continue it.

File: source/includes/rust/hooks_structure.md [hunk 4]
```diff
@@ -36,6 +36,19 @@ void CanChangeCode(CodeLock codeLock, BasePlayer player, string newCode, bool is
...
+
+``` csharp
+void CanPickupLock(BasePlayer player, BaseLock lock)
+{
+    Puts("CanPickupLock works!");
+}
+```
+
+ * Called when a player attempts to pickup a lock
+ * Returning true or false overrides default behavior
```
Comment (Mughisi) [inline source/includes/rust/hooks_structure.md]: Returning a non-null value will override the default behavior, returning true or false would always interrupt it.

File: source/includes/rust/hooks_weapon.md [hunk 5]
```diff
@@ -71,3 +71,27 @@ void OnWeaponFired(BaseProjectile projectile, BasePlayer player, ItemModProjecti
 
  * Called when the player fires a weapon
  * No return behavior
+
+## OnReloadMagazine
+
+``` csharp
+void OnReloadMagazine(BaseProjectile projectile, BasePlayer player, ItemModProjectile mod, ProtoBuf.ProjectileShoot projectiles)
```
Comment (Mughisi) [inline source/includes/rust/hooks_weapon.md]: OnReloadMagazine(BasePlayer player, BaseProjectile projectile)

File: source/includes/rust/hooks_weapon.md [hunk 6]
```diff
@@ -71,3 +71,27 @@ void OnWeaponFired(BaseProjectile projectile, BasePlayer player, ItemModProjecti
...
+}
+```
+
+ * Called when the player reloads a magazine
+ * Returning a non-null value overrides default behavior
+ 
+## OnReloadWeapon
+
+``` csharp
+void OnReloadWeapon(BaseProjectile projectile, BasePlayer player, ItemModProjectile mod, ProtoBuf.ProjectileShoot projectiles)
```
Comment (Mughisi) [inline source/includes/rust/hooks_weapon.md]: OnReloadWeapon(BasePlayer player, BaseProjectile projectile)

**Comment by lukespragg** [comment]
Manually merged with requested changes.

**Pull request closed without merging**